src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -751,6 +751,81 @@ public void PresentChain_CheckpointsEnabled_MarkToDownloadWhenTwoCheckpointsPres
+        {
+            // Checkpoints are disabled.
+            // Initial chain has headers (h1-h10).
+            const int initialChainOfTenBlocks = 10;
+            TestContext testContext = new TestContextBuilder().WithInitialChain(initialChainOfTenBlocks)
+                .UseCheckpoints(false).Build();
+            ChainedHeaderTree chainedHeaderTree = testContext.ChainedHeaderTree;
+            ChainedHeader initialChainTip = testContext.InitialChainTip;
+
+            // Assume valid at Block X (h30).
+            const int assumeValidBlockHeight = 30;
+            const int extendChainByTwentyBlocks = 20;
+            ChainedHeader extendedChainTip = testContext.ExtendAChain(extendChainByTwentyBlocks, initialChainTip);
+            Assert.Equal(extendedChainTip.Height, assumeValidBlockHeight);
+            testContext.ConsensusSettings.BlockAssumedValid = extendedChainTip.HashBlock;
+            
+            // Blocks up to X-10 (h11->h20) are marked for download.
+            List<BlockHeader> listOfCurrentChainHeaders =
+                testContext.ChainedHeaderToList(extendedChainTip, initialChainOfTenBlocks + extendChainByTwentyBlocks).Take(extendChainByTwentyBlocks).ToList();
+            ConnectNewHeadersResult connectNewHeadersResult = chainedHeaderTree.ConnectNewHeaders(1, listOfCurrentChainHeaders);
+            ChainedHeader chainedHeaderDownloadFrom = connectNewHeadersResult.DownloadFrom;
+            ChainedHeader chainedHeaderDownloadTo = connectNewHeadersResult.DownloadTo;
+
+            chainedHeaderDownloadFrom.HashBlock.Should().Be(initialChainTip.Next[0].HashBlock); // h11
+            chainedHeaderDownloadTo.HashBlock.Should().Be(listOfCurrentChainHeaders.Last().GetHash());
+
+            // Blocks before X-20 (h1->h10) are FV.
+            ChainedHeader chainedHeader = initialChainTip;
+            Assert.Equal(chainedHeader.Height, assumeValidBlockHeight - 20);
+            ValidationState expectedState = ValidationState.FullyValidated;
+            while (chainedHeader.Height > 0)
+            {
+                chainedHeader.BlockValidationState.Should().Be(expectedState);
+                chainedHeader = chainedHeader.Previous;
+            }
+
+            // Headers up to block X (h30) + some more (h31->h35) are presented.
+            const int extendChainByFiveBlocks = 5;
+            extendedChainTip = testContext.ExtendAChain(extendChainByFiveBlocks, extendedChainTip);
+            listOfCurrentChainHeaders =
+                testContext.ChainedHeaderToList(extendedChainTip, assumeValidBlockHeight + extendChainByFiveBlocks);
+            connectNewHeadersResult = chainedHeaderTree.ConnectNewHeaders(1, listOfCurrentChainHeaders);
+
+            // All from X-10 (h21->h30) are marked for download.
+            ChainedHeader fromHeader = chainedHeaderTree.GetChainedHeadersByHash().Values.First(x => x.Height > assumeValidBlockHeight - 10);
```

Comment:
Use get ancestor. Dictionaries are not sorted, hence first has little meaning.

Reply:
Done

Reply:
I still see the old code here.

Reply:
Still doesn't seem to be changed.

Reply:
Should be sorted now.